app/src/main/java/co/techmagic/hr/presentation/time_tracker/HrAppTimeTrackerPresenter.kt

```diff
@@ -194,9 +194,16 @@ class HrAppTimeTrackerPresenter(
             timeTrackerInteractor.stopTimer().subscribe({ updateReportViewModel(TaskUpdate(it, TaskTimerState.RUNNING)) }, this::showError)
         } else {
             if (canStartReportTracking(userReportViewModel)) {
-                val userReport = userReportViewMadelMapper.retransform(userReportViewModel)
-                timeTrackerInteractor.startTimer(userReport, totalDayMinutesExcludeCurrent(userReport.date.toCalendar(), userReport.minutes))
-                        .subscribe({ updateReportViewModel(TaskUpdate(it.current, TaskTimerState.RUNNING)) }, this::showError)
+                router?.let {
+                    if (!it.isBatteryOptimizationTurnedOn()) {
```

Comment:
As I understood, we should handle this in all places where we're trying to start a timer, right?
Does it make sense to move this check to the service into `startTimer` method and throw an Exception if optimization is turned on? We can handle our custom exception right in the base fragment in `handleError` method.

Reply:
Agree